diff --git a/Makefile b/Makefile
index abce984..5a40d01 100644
--- a/Makefile
+++ b/Makefile
@@ -66,7 +66,9 @@ go.mod:
 	$(GO) mod init github.com/sonic-net/sonic-gnmi
 
 $(GO_DEPS): go.mod $(PATCHES) swsscommon_wrap $(GNOI_YANG)
-	$(GO) mod vendor
+	GOPRIVATE= GOPROXY=https://proxy.golang.org,direct $(GO) mod vendor || \
+	GOPRIVATE= GOPROXY=https://mirrors.aliyun.com/goproxy/,direct $(GO) mod vendor || \
+	GOPRIVATE= GOPROXY=direct $(GO) mod vendor
 	$(GO) mod download github.com/google/gnxi@v0.0.0-20181220173256-89f51f0ce1e2
 	cp -r $(GOPATH)/pkg/mod/github.com/google/gnxi@v0.0.0-20181220173256-89f51f0ce1e2/* vendor/github.com/google/gnxi/
 
